ts/state/ducks/onion.tsx

```diff
@@ -0,0 +1,41 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import _, { forEach } from 'underscore';
+import { SnodePath, Snode } from '../../session/onions/index';
+
+export type OnionState = {
+  // nodes: Array<OnionPathNodeType>;
+  // path: SnodePath;
+  snodePath: SnodePath;
+};
+
+// const initialState: OnionState = {
+//   // nodes: new Array<OnionPathNodeType>(),
+//   nodes: new Array<Snode>(),
+// };
+
+const initialState = {
+  snodePath: {
+    path: new Array<Snode>(),
+    bad: false,
+  },
+};
+
+/**
+ * This slice is the one holding the default joinable rooms fetched once in a while from the default opengroup v2 server.
+ */
+const onionSlice = createSlice({
+  name: 'onionPaths',
+  initialState,
+  reducers: {
+    updateOnionPaths(state, action: PayloadAction<SnodePath>) {
+      let newPayload = { snodePath: action.payload };
+      let isEqual = JSON.stringify(state, null, 2) == JSON.stringify(newPayload, null, 2);
+      return isEqual ? state : newPayload;
+    },
```

Comment:
createSlice take care of not triggering update if the state did not change. 
So you can basically return `{ snodePath: action.payload }` directly if I get this correctly. No need to compare it.

and If you need to compare records, no need to put them in json. Use `_.deepEqual` from `lodash`